crates/migrations/src/utils/migration_utils.rs

```diff
@@ -0,0 +1,111 @@
+use std::fs::File;
+use std::io::Result;
+use std::path::PathBuf;
+use std::time::SystemTime;
+use tar::Builder;
+
+// Utility method used to create a tar file from the specified directory.
+// The generated tar file will have the same name as the directory with
+// the utc timestamp and the extension "old" append to it.
+pub fn backup_dir(dir: &PathBuf) -> Result<()> {
+    let prefix = dir.file_name().unwrap();
+    let time_str = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_millis()
+        .to_string();
+    // Name of output tar file
+    let output_file = format!("{}.{}.{}", prefix.to_str().unwrap(), time_str, "old");
+
+    let logging_str = output_file.clone();
+
+    // Full path to output tar file
+    let backup_path = dir.parent().unwrap().join(output_file);
+
+    let file_result = File::create(backup_path);
+    if let Err(e) = file_result {
+        println!(
+            "Error generating backup file {:?} for directory {:?}, Error: {:?}",
+            logging_str, dir, e
+        );
+        return Err(e);
+    }
+
+    let file = file_result.unwrap();
+    let mut tar_builder = Builder::new(file);
+
+    let tar_response = tar_builder.append_dir_all(".", dir);
+    if let Err(e) = tar_response {
+        println!("Error adding files to tar, {:?}", e);
+        return Err(e);
+    }
+    return Ok(());
```

Comment:
clippy (`cargo clippy`, the built-in Rust linter) will complain about this, replace with `Ok(())`

Reply:
Updated and included switching if let statement with ?